### ISSUE TITLE
Navigation of Lemmas, Definitions, etc. using DocumentSymbols

### ISSUE THREAD

**Issue by RobinTruax** [latex-lsp/texlab#893]
This is related to the resolved discussion in #31 concerning DocumentSymbols which included mentions of the table of contents functionality of `vimtex`, as well as equations/lemmas/definitions/theorems/etc. I'm trying to recreate an fancy LSP pop-up version of the `vimtex` table of contents in Neovim. 

I'm using [`nvim-navic`](https://github.com/SmiteshP/nvim-navic) and [`nvim-navbuddy`](https://github.com/SmiteshP/nvim-navbuddy); the former reads the navigation data from the LSP using `DocumentSymbols`, and the latter provides the pop-up menu. For the sake of this discussion, only the former matters. 

I've gotten `nvim-navic` working with `texlab`, but it's only recognizing sections, subsections, equations, and enumerate objects. For me, this doesn't provide a very useful navigational system; I'd like to have a list of sections, subsections, and then the usual environments which I define in my standard style file: definitions, theorems, lemmas, corollaries, propositions, etc., tagged with their number and title. 

I understand that creating this this is half a `texlab` issue and half a `nvim-navic` issue, so primarily my question has to do with input: how can one obtain the required list of data to pipe into `nvim-navic` to make this happen? Happy to provide screenshots of the current system and/or mockups of what I'm trying to create if it'd help.

**Comment by clason** [comment]
I don't understand; texlab returns symbols for all kinds of objects, including theorems, lemmas, definitions. See https://github.com/latex-lsp/texlab/wiki/LSP-Internals#enum-mapping

**Comment by RobinTruax** [comment]
You're right. The issue is resolved when everything is in a single file; screenshot attached, see winbar. It looks like my theorem and definitions environments aren't being recognized because they're in a different style file (I have a standard one for all my notes). Do you know how to make texlab recognize the content of those style files?

![image](https://github.com/latex-lsp/texlab/assets/40613148/cab7020c-253b-482d-ab36-47f86608064d)

**Comment by clason** [comment]
Are you sure you're using `workspace/symbol` and not `textDocument/documentSymbol` (which is supposed to only return symbols from the current document according to the spec)?

**Comment by RobinTruax** [comment]
Thanks for that hint. I was able to resolve the first issue (regarding external files) with configuration and verified that it worked using a minimal example, but my style file still wasn't working. 

After some A-B testing, I figured out why, and I do actually think this is the root of the issue: theorem-style environments defined using `thmtools` are not recognized by the LSP; only theorem-style environments defined by `amsthm`. Consider the following two examples: 

![image](https://github.com/latex-lsp/texlab/assets/40613148/06191d33-dbd5-492a-990c-89b9e24c2760)
![image](https://github.com/latex-lsp/texlab/assets/40613148/41fd043e-f685-4eeb-bb00-0f928faeff94)
